crates/glue/src/types.rs

```diff
@@ -1196,6 +1212,35 @@ impl<'a> Env<'a> {
         }
     }
 
+    fn add_thunk_entry_point(
+        &mut self,
+        types: &mut Types,
+        name: String,
+        ret_type_id: TypeId,
+    ) -> TypeId {
+        let thunk_id = {
+            let typ = RocType::Function(RocFn {
+                // TODO investigate: is LambdaSetId(1) always correct here?
+                // It's supposed to be the LambdaSetId for this thunk,
+                // but what LambdaSetId should that be, if not 1?
+                extern_name: fn_caller_name(&name, LambdaSetId(1)),
```

Comment:
this should indeed be correct because thunks only dispatch to one destination, and more over only have place in their type that they can dispatch